script.module.clouddrive.common/clouddrive/common/service/base.py

```diff
@@ -0,0 +1,138 @@
+            Logger.notice('Service stopped')
+    
+class BaseServer(ThreadingTCPServer):
+    
+    data = None
+    service = None
+    allow_reuse_address = True
+    
+    def __init__(self, server_address, RequestHandlerClass, service, data=None):
+        self.data = data
+        self.service = service
+        #HTTPServer.__init__(self, server_address, RequestHandlerClass)
+        ThreadingTCPServer.__init__(self, server_address, RequestHandlerClass)
+    
+    def process_request(self, request, client_address):
+        thread = threading.Thread(target = self.process_request_thread, args = (request, client_address), name='%s-request' % threading.current_thread().name)
+        thread.daemon = True
+        thread.start()
+
+class BaseHandler(BaseHTTPRequestHandler):
+    content_type = 'text/html; charset=UTF-8'
+    response_code_sent = False
+    response_headers_block_sent = False
+    
+    def __init__(self, request, client_address, server):
+        self.protocol_version = 'HTTP/1.1'
+        self.server_version = KodiUtils.get_addon_info('id') + '/' + KodiUtils.get_addon_info('version')
+        self.wbufsize = -1
+        BaseHTTPRequestHandler.__init__(self, request, client_address, server)
+    
+    def write_response(self, code, message=None, content=None, headers=None):
+        headers = Utils.default(headers, {})
+        length = 0
+        self.send_response(code, message)
+        if content:
+            length = content.tell()
+            content.seek(0)
+            self.send_header('Content-Type', self.content_type)
+            self.send_header('Content-Length', length)
+        self.send_header('Connection', 'close')
+        request_id = self.headers.get('request-id')
+        if request_id:
+            self.send_header('request-id', request_id)
+        for key in headers:
+            self.send_header(key, headers[key])
+        self.end_headers()
+        if self.command != 'HEAD' and code >= 200 and code not in (204, 304):
+            if content:
+                shutil.copyfileobj(content, self.wfile)
+                content.close()
+            Logger.debug('[%s] response code: %s, length: %s' % (request_id, code, length))
+    
+    def send_response(self, code, message=None):
+        if self.response_code_sent:
+            raise InvalidResponseException('response code already sent')
+        BaseHTTPRequestHandler.send_response(self, code, message)
+        self.response_code_sent = True
+        
+    def send_header(self, keyword, value):
+        if self.response_headers_block_sent:
+            raise InvalidResponseException('Response headers block already sent')
+        BaseHTTPRequestHandler.send_header(self, keyword, value)
+    
+    def end_headers(self):
+        BaseHTTPRequestHandler.end_headers(self)
+        self.response_headers_block_sent = True
+     
+    def log_message(self, format, *args):
+        #Logger.notice("[%s.service] %s\n" % (self.server.service.name, format%args))
```

Comment:
intentional?

Reply:
was intentional, but it's a good time to remove these unused methods. done.